### DISCUSSION TITLE
I was hoping to download the built ipa for my ipad to install using troll store

### DISCUSSION THREAD

**Discussion by Infiniteking21** [yuroyami/syncplay-mobile#96] [Q&A] [answered]
I saw that there was no ipa so i tried to get informed on how to build it myself, I have no experience there and i run linux with no mac, i had to figure out how to setup a vm with sonoma on it, i got xcode and tried to just open the project file from the zip and hit build but i get hit with tons of errors a lot fo which relating to it trying to point to directories that dont exist anymore in the project. I was wondering if someone could just help me out with the ipa instead, I would really appreciate it, thanks!

**Comment by yuroyami** [comment]
@Infiniteking21 It may be really challenging to build the entire thing by yourself on your own Mac, so I recommend you not bother with it anymore because it would take you a lot of time and confusion. 
For the IPA, I am currently swamped and busy as my plate is full. I will get around to providing the IPA very soon. I will also try to make the app obtainable via third-party stores like AltStore etc.

**Reply by Infiniteking21** [reply]
Thank you for the reply your work on this app is great, I tested it on an android device already so ill be looking forward to that ios version. I appreciate it thanks.

**Reply by Amadace** [reply]
I downloaded your project and attempted to run it on my device, but I get an error about Pods.
I installed Pods but upon running pod install, I am told to run:
./gradlew :shared:generateDummyFramework

Upon running:
../gradlew :shared:generateDummyFramework 
(with 2 dots at the start, as gradlew is the the previous directory)

I get, Could not find or load main class org.gradle.wrapper.GradleWrapperMain

I try to fix this by running:
gradle wrapper

But that I get

Project directory '/Users/ace/Documents/syncplay-mobile-0.14.0/iosApp' is not part of the build defined by settings file '/Users/ace/Documents/syncplay-mobile-0.14.0/settings.gradle.kts'. If this is an unrelated build, it must have its own settings file.

Any help on this please?

**Reply by yuroyami** [reply]
@Amadace Building a KMP project can be a real pain in the ass as I'd expect it to throw a million errors my way on a fresh environment. I suggest grabbing the code again from the release assets of `0.14.0`. Don't open it or do any building yet, follow this:

1- I am running a macOS (which is necessary for building for iOS) Ventura, `Android Studio Koala | 2024.1.1 Beta 2` and `Xcode 15.2 (with a simulator runtime for iOS 17.2)` (I believe the only thing necessary in this case is having both Android Studio and Xcode, versions shouldn't matter much but it's always advisable to grab the latest versions).
2- I have `cocoapods` installed via Homebrew: `brew install cocoapods`
3- I make sure Android Studio has this plugin installed: [KMM plugin](https://plugins.jetbrains.com/plugin/14936-kotlin-multiplatform-mobile)
4- I have `[kdoctor](https://github.com/Kotlin/kdoctor)` installed via Homebrew (this tool allows you to check whether your environment is ready for KMP builds).  You can run the command `kdoctor` which should tell you whether something is missing.
5- For the missing `Gradle wrapper`, I have it compressed for you and attached to this. Simply unzip it at the root directory so that the `gradle` folder sits beside other folders like `androidApp`, `iosApp` etc.
6- Now open the project in Android Studio and if it starts to sync the project automatically, stop it. Go to `shared/build.gradle.kts` and comment the lines that specify the two pods (one for VLC and one for SPLPing, you should fine these two lines within the `cocoapods { }` lambda). 
7- Sync the project now. It will download all the necessary libraries. Once finished, you can uncomment the two pod lines again.
8- Now open the xcworkspace project in Xcode, build it.
9- At this point, everything should be built correctly and the app should be able to run on your device with no issues. 

[gradle.zip](https://github.com/yuroyami/syncplay-mobile/files/15478960/gradle.zip)

It is very likely that you'd encounter other errors along the way when building it so I suggest to wait for the app store release sometimes this June.

**Reply by N0word** [reply]
I used that method but failed, it shows:
`> Task :shared:cinteropMobileVLCKitIosArm64 FAILED
Exception in thread "main" java.lang.Error: /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.0.sdk/usr/include/copyfile.h:35:10: fatal error: 'sys/cdefs.h' file not found
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.0.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:75:10: fatal error: could not build module 'Darwin'
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.0.sdk/usr/include/dispatch/dispatch.h:28:10: fatal error: could not build module 'Darwin'
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.0.sdk/usr/include/os/object.h:28:10: fatal error: could not build module 'Darwin'
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.0.sdk/usr/include/objc/objc-auto.h:36:10: fatal error: could not build module 'Darwin'
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.0.sdk/usr/include/os/workgroup_base.h:15:10: fatal error: could not build module 'Darwin'
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.0.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:6:10: fatal error: could not build module 'CoreFoundation'
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.0.sdk/System/Library/Frameworks/Security.framework/Headers/SecBase.h:28:10: fatal error: could not build module 'CoreFoundation'
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.0.sdk/System/Library/Frameworks/CFNetwork.framework/Headers/CFNetwork.h:18:10: fatal error: could not build module 'CoreFoundation'
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS18.0.sdk/usr/include/xpc/xpc.h:6:10: fatal error: could not build module 'os_object'
	at org.jetbrains.kotlin.native.interop.indexer.ModuleSupportKt.getModulesASTFiles(ModuleSupport.kt:80)
	at org.jetbrains.kotlin.native.interop.indexer.ModuleSupportKt.getModulesInfo(ModuleSupport.kt:15)
	at org.jetbrains.kotlin.native.interop.gen.jvm.MainKt.buildNativeLibrary(main.kt:564)
	at org.jetbrains.kotlin.native.interop.gen.jvm.MainKt.processCLib(main.kt:308)
	at org.jetbrains.kotlin.native.interop.gen.jvm.MainKt.processCLibSafe(main.kt:244)
	at org.jetbrains.kotlin.native.interop.gen.jvm.MainKt.access$processCLibSafe(main.kt:1)
	at org.jetbrains.kotlin.native.interop.gen.jvm.Interop.interop(main.kt:102)
	at org.jetbrains.kotlin.cli.utilities.InteropCompilerKt.invokeInterop(InteropCompiler.kt:49)
	at org.jetbrains.kotlin.cli.utilities.MainKt.mainImpl(main.kt:23)
	at org.jetbrains.kotlin.cli.utilities.MainKt.main(main.kt:44)
error: Process 'command '/usr/local/Cellar/openjdk@17/17.0.12/libexec/openjdk.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':shared:cinteropMobileVLCKitIosArm64'.
> Process 'command '/usr/local/Cellar/openjdk@17/17.0.12/libexec/openjdk.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.9/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 38s
7 actionable tasks: 1 executed, 6 up-to-date`

**Reply by N0word** [reply]
How could I solve it? THX!

**Reply by yuroyami** [reply]
@N0word It's hard to tell what exactly is going wrong. But given the fact that it is failing on the task `> Task :shared:cinteropMobileVLCKitIosArm64 FAILED`, then I suppose it has to do with Cocoapods. I suggest you run a `kdoctor` to see if your environment is completely set up for Kotlin multiplatform development and building. 

You can find `kdoctor` [here](https://github.com/Kotlin/kdoctor). To sum it up for you, you just need to install it via Homebrew with the command `brew install kdoctor`, then after it's installed just use the command `kdoctor` and it will check all the requisites. Please post the output here and I'll try to help further

**Reply by N0word** [reply]
Yeah, I also used the `kdoctor`, and there's no problems :
Environment diagnose (to see all details, use -v option):
[✓] Operation System
[✓] Java
[✓] Android Studio
[✓] Xcode
[✓] CocoaPods

Conclusion:
  ✓ Your operation system is ready for Kotlin Multiplatform Mobile Development!

**Reply by yuroyami** [reply]
@N0word 
For troubleshooting purposes, what happens when you try to sync the project when you comment out the lines of `pod("MobileVLCKit")` in the `cocoapods { }` block inside `build.gradle.kts` ? The project should sync fine if there is no issue with the whole environment.

**Reply by N0word** [reply]
Emmm…those lines were commented out before the sync, and the process worked well. Maybe I should try agian?

**Comment by yuroyami** [comment, marked as answer]
The `ipa` package is now attached with the `0.14.0` release. I am not sure if I have generated the `ipa` correctly but hopefully it works. Feel free to share the results here!

**Reply by Infiniteking21** [reply]
thanks! it worked great im able to watch anime with my freind from my ipad now.

**Reply by bachig26** [reply]
> thanks! it worked great im able to watch anime with my freind from my ipad now.

Have you tried with Altstore, it's not working for me.

**Comment by yuroyami** [comment]
An `.ipa` file is attached to every new release.
